pulumi/python/automation/main.py

```diff
@@ -85,28 +96,45 @@ def usage():
     print(usage_text, file=sys.stdout)
 
 
-def provider_instance(provider_name: str) -> Provider:
-    """Dynamically instantiates an infrastructure provider
-    :param provider_name: name of infrastructure provider
-    :return: instance of infrastructure provider
-    """
-    module = importlib.import_module(name=f'providers.{provider_name}')
-    return module.INSTANCE
+def write_env(env_config, stack_name):
+    """Create a new environment file and write our stack to it"""
+    with open(env_config.filename, 'w') as f:
+        try:
+            print("PULUMI_STACK=" + stack_name, file=f)
+            msg = 'Environment configuration file not found. Creating new file at the path: %s'
+            RUNNER_LOG.error(msg, env_config.filename)
```

Comment:
This section is a little confusing since `print` is being used as the way to write to the file.  Although it seems this is reasonably common in python I'd suggest a comment on print to clarify what is going on.

One other small nit, is the fact that a new file was created an `error`?  Seems like in the logs this would be an `info` level message?

Reply:
Added comments before each print statement to make it clear we are writing to a file, and adjusted the log level to info.